v2/dtos/requests/provisionwatcher_test.go

```diff
@@ -0,0 +1,328 @@
+//
+// Copyright (C) 2021 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package requests
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var testProvisionWatcherName = "TestWatcher"
+var testProvisionWatcherLabels = []string{"TEST", "TEMP"}
+var testIdentifiers = map[string]string{
+	"address": "localhost",
+	"port":    "3[0-9]{2}",
+}
+var testBlockingIdentifiers = map[string][]string{
+	"port": {"397", "398", "399"},
+}
+var testAddProvisionWatcher = AddProvisionWatcherRequest{
+	BaseRequest: common.BaseRequest{
+		RequestId: ExampleUUID,
+	},
+	ProvisionWatcher: dtos.ProvisionWatcher{
+		Name:                testProvisionWatcherName,
+		Labels:              testProvisionWatcherLabels,
+		Identifiers:         testIdentifiers,
+		BlockingIdentifiers: testBlockingIdentifiers,
+		ServiceName:         TestDeviceServiceName,
+		ProfileName:         TestDeviceProfileName,
+		AdminState:          models.Locked,
+		AutoEvents:          testAutoEvents,
+	},
+}
+
+var testUpdateProvisionWatcher = UpdateProvisionWatcherRequest{
+	BaseRequest: common.BaseRequest{
+		RequestId: ExampleUUID,
+	},
+	ProvisionWatcher: mockUpdateProvisionWatcher(),
+}
+
+func mockUpdateProvisionWatcher() dtos.UpdateProvisionWatcher {
+	testId := ExampleUUID
+	testName := testProvisionWatcherName
+	testAdminState := models.Locked
+	testDeviceServiceName := TestDeviceServiceName
+	testProfileName := TestDeviceProfileName
+	d := dtos.UpdateProvisionWatcher{}
+	d.Id = &testId
+	d.Name = &testName
+	d.Labels = testProvisionWatcherLabels
+	d.Identifiers = testIdentifiers
+	d.BlockingIdentifiers = testBlockingIdentifiers
+	d.ServiceName = &testDeviceServiceName
+	d.ProfileName = &testProfileName
+	d.AdminState = &testAdminState
+	d.AutoEvents = testAutoEvents
+	return d
+}
+
+func TestAddProvisionWatcherRequest_Validate(t *testing.T) {
+	emptyString := " "
+	emptyMap := make(map[string]string)
+	valid := testAddProvisionWatcher
+	noReqId := testAddProvisionWatcher
+	noReqId.RequestId = ""
+	invalidReqId := testAddProvisionWatcher
+	invalidReqId.RequestId = "abc"
+	noProvisionWatcherName := testAddProvisionWatcher
```

Comment:
Add one for invalid character(s) in name. i.e. edgex-dto-rfc3986-unreserved-chars fails.